csharp/ql/src/semmle/code/csharp/dataflow/LibraryTypeDataFlow.qll

```diff
@@ -721,6 +721,42 @@ class SystemLazyFlow extends LibraryTypeDataFlow, SystemLazyClass {
   }
 }
 
+/** Data flow for `System.Nullable<>`. */
+class SystemNullableFlow extends LibraryTypeDataFlow, SystemNullableStruct {
+  override predicate callableFlow(
+    CallableFlowSource source, AccessPath sourceAp, CallableFlowSink sink, AccessPath sinkAp,
+    SourceDeclarationCallable c, boolean preservesValue
+  ) {
+    preservesValue = true and
+    c.(Constructor).getDeclaringType() = this and
+    source = getFlowSourceArg(c, 0, sourceAp) and
+    sourceAp = AccessPath::empty() and
+    sink = TCallableFlowSinkReturn() and
+    sinkAp = AccessPath::property(this.getValueProperty())
+    or
+    preservesValue = true and
+    c = this.getAGetValueOrDefaultMethod() and
+    source = TCallableFlowSourceQualifier() and
+    sourceAp = AccessPath::property(this.getValueProperty()) and
+    sink = TCallableFlowSinkReturn() and
+    sinkAp = AccessPath::empty()
+    or
+    preservesValue = false and
+    c = this.getHasValueProperty().getGetter() and
+    source = TCallableFlowSourceQualifier() and
+    sourceAp = AccessPath::property(this.getValueProperty()) and
+    sink = TCallableFlowSinkReturn() and
+    sinkAp = AccessPath::empty()
+    or
+    preservesValue = true and
+    c = this.getAGetValueOrDefaultMethod() and
+    source = getFlowSourceArg(c, 0, _) and
+    sourceAp = AccessPath::empty() and
+    sink = TCallableFlowSinkReturn() and
+    sinkAp = AccessPath::empty()
+  }
```

Comment:
I think we should also add
```ql
    or
    preservesValue = false and
    c = this.getValueProperty().getGetter() and
    source = TCallableFlowSourceQualifier() and
    sourceAp = AccessPath::empty() and
    sink = TCallableFlowSinkReturn() and
    sinkAp = AccessPath::empty()
```

This will allow taint to propagate from `Nullable<T>` objects to their `Value` property, which might be useful when specifying a `Nullable<T>` object as a source in a taint-tracking configuration. (We should do the same for the `Value` property of `Lazy<T>` objects.)

Reply:
Does this basically say that the nullable object and its `Value` property have the same value?